### ISSUE TITLE
[Data Store API] Add node without adding collection

### ISSUE THREAD

**Issue by Tummerhore** [gridsome/gridsome#1007]
I'm not quite sure if this already possible. But as I did not find a solution I'm posting this as a feature request.
It would be nice to be able to add nodes to the GraphQl Layer without adding a collection first. Many headless CMS have a concept of singletons, which I would like to map to top level GraphQl fields. There is no need to have the automatic creation of an additional "all..." field as there is only one item. But it seems like the data store api only allows adding nodes to collections.

Thanks for reading.

**Comment by noxify** [comment]
Hi @Tummerhore,

is there an example/use case for this?
Just to make sure, we have the same understanding.

Thanks.

**Comment by Tummerhore** [comment]
Yeah, sure. I want to add the Collections and Singletons from Cockpit CMS to GraphQL in Gridsome. According to the documentation I can use the Data Store API for that. For example:
```
api.loadSource(actions => {
  const posts = actions.addCollection({
    typeName: 'BlogPost'
  })

  posts.addNode({
    title: 'My first blog post',
    date: '2018-11-02',
    customField: 'My value'
  })
  posts.addNode(<...>)
})
```
This will lead to the following GraphQl fields: _allBlogPost_ and _BlogPost_, where the first one is an array of BlogPosts and the later one is a single entry but requires an attribut (id or path).

This is fine for my Cockpit collections. But now, I also want to add Cockpit singletons. These are pages where only one item exists, e.g. I have a singleton for the homepage of a website. I would like to have one GraphQl field to easily access the singleton data like this:
```
query {
  homepage {
    title,
    picture,
    content
  }
}
```
But as the data store api only allows adding collections to the GraphQl top level (at least as far as I know), there will be two fields again: _allHomepage_ and _homepage_ and I have to access the homepage data through an array which only has one element, thus the array is completely unncessary and just complicates the query. The other created field _homepage_ can only be used with additional arguments to identfiy the item in the array. Again, this is unnecessarily complicated as the array can only consist of one item.
So, in my opinion it would be a good idea to allow something like the following, which would only create one GraphQL field, which can be queried like shown above:
```
api.loadSource(actions => {
  actions.addSingleton({
      typeName: 'homepage'
    }, {
      title: <...>
      picture: <...>
      content: <...>
    }
  )
})
```

**Comment by hacknug** [comment]
Wouldn't adding a Singletons collection with all of them as nodes solve (at least partially) your issue?

**Comment by Tummerhore** [comment]
Yes, that's exactly what I did in the end. But in my opinion it would be nicer to have them as a seperate field at the top level in the future as in most cases you do not iterate through all available Singletons (like for the children of collections). That's why I made this feature request.

**Comment by lipengzhou** [comment]
I have the same functional needs as he does.
